stl/src/vector_algorithms.cpp

```diff
@@ -433,6 +439,626 @@ __declspec(noalias) void __cdecl __std_reverse_copy_trivially_copyable_8(
+
+                        if (_Mode == _Min_max_mode::_Both ? _Cur_max_val <= _H_max_val : _Cur_max_val < _H_max_val) {
+                            // max_element: current horizontal max is greater than the old, update max
+                            // minmax_element: current horizontal max is not less than the old, update max
+                            _Cur_max_val = _H_max_val;
+                            const __m128i _Eq_mask =
+                                _Traits::_Cmp_eq(_H_max, _Cur_vals_max); // Mask of all elems eq to max
+                            int _Mask = _mm_movemask_epi8(_Eq_mask);
+
+                            unsigned long _H_pos;
+                            if constexpr (_Mode == _Min_max_mode::_Both) {
+                                // Looking for the last occurence of maximum
+                                // Indices of maximum elements or zero if none
+                                const __m128i _Idx_max_val =
+                                    _mm_blendv_epi8(_mm_setzero_si128(), _Cur_idx_max, _Eq_mask);
+                                const __m128i _Idx_max = _Traits::_H_max_u(_Idx_max_val); // The greatest indices
+                                // Select the greatest vertical indices from the largest element mask
+                                _Mask &= _mm_movemask_epi8(_Traits::_Cmp_eq(_Idx_max, _Idx_max_val));
+                                _BitScanReverse(&_H_pos, _Mask); // Find the largest horizontal index
+                                _H_pos -= sizeof(_Cur_max_val) - 1; // Correct from highest val bit to lowest
+                            } else {
+                                // Looking for the first occurence of maximum
+                                // Indices of maximum elements or the greatest index if none
+                                const __m128i _All_max     = _mm_set1_epi8(static_cast<char>(0xFF));
+                                const __m128i _Idx_max_val = _mm_blendv_epi8(_All_max, _Cur_idx_max, _Eq_mask);
+                                const __m128i _Idx_max     = _Traits::_H_min_u(_Idx_max_val); // The smallest indices
+                                // Select the smallest vertical indices from the largest element mask
+                                _Mask &= _mm_movemask_epi8(_Traits::_Cmp_eq(_Idx_max, _Idx_max_val));
+                                _BitScanForward(&_H_pos, _Mask); // Find the smallest horizontal index
+                            }
+
+                            const auto _V_pos = _Traits::_Get_v_pos(_Cur_idx_max, _H_pos); // Extract its vertical index
+                            _Res._Max         = _Base + _V_pos * 16 + _H_pos; // Finally, compute the pointer
+                        }
+                    }
+                    // Horizontal part done, results are saved, now need to see if there is another portion to process
+
+                    if constexpr (_Traits::_Has_portion_max) {

```

Comment:
some of this stuff could perhaps be factored out and shared with the pre-loop initialization above, but I'm not sure that would actually improve clarity here.

Reply:
It is not easy from control flow perspective, unless we can use `goto`, as it involves breaking out of inner loop.

Reply:
I'm actually OK with goto if the improvement in clarity is worth it. I'm especially OK with using it as a "labeled break" (even in C where goto can be UB I'm almost always OK with it, C's goto is _nothing_ like the goto dijkstra was complaining about, and tbh I think ignoring dijkstra is pretty frequently a good idea).

I can't speak for the other maintainers though.

Reply:
I assumed initially that the (re-)initialization will be put just above `for(;;)` so that the label itself would be a loop.

But there's labelled break solution either: add outer `do{}` while(false)`, put `continue` here, and make the two `breaks` below labelled breaks of that outer loop.

Maybe I'm missing a simpler solution though.

Reply:
I'm ok with the control flow as-is. We've avoided `goto` in the codebase and I wouldn't be eager to see it introduced.